### PR TITLE
Cast feature

### PR DESCRIPTION
**Pull request by reuster986**
This PR allows users to explicitly cast arrays between dtypes. Addresses #380

### CODE REVIEW

File: arkouda/numeric.py [hunk 1]
```diff
@@ -3,13 +3,56 @@
...
+    has the precision and capacity to hold the desired result.
+    """
+
+    if isinstance(pda, pdarray):
+        name = pda.name
+        objtype = "pdarray"
+    elif isinstance(pda, Strings):
+        name = '+'.join((pda.offsets.name, pda.bytes.name))
+        objtype = "str"
+    else:
```
Comment (hokiegeek2) [inline arkouda/numeric.py:44]: This clause is not needed as @typechecked confirms the pda is a pdarray or Strings

File: arkouda/numeric.py [hunk 2]
```diff
@@ -3,13 +3,56 @@
...
+        raise TypeError("Input must be pdarray or Strings")
+
+    dt = _as_dtype(dt)
+    opt = ""
+    msg = "cast {} {} {}".format(name, objtype, dt.name, opt)
```
Comment (mhmerrill) [inline arkouda/numeric.py:49]: needs another format field for `opt`

File: arkouda/numeric.py (hunk 1 above)
Comment (reuster986) [reply arkouda/numeric.py]: Good reminder! Still getting used to it :) Fixed just now.

File: arkouda/numeric.py (hunk 2 above)
Comment (reuster986) [reply arkouda/numeric.py]: Fixed.

**Review by hokiegeek2 — APPROVED** [review]

**Review by mhmerrill — APPROVED** [review]
looks good to me

**Pull request merged**